Artifacts/windows-azuredevops-multiple-agents/run.ps1

```diff
@@ -0,0 +1,122 @@
+# Downloads the Azure DevOps Pipelines Agent and installs specified instances on the new machine
+# under C:\agents\ and registers with the Azure DevOps Pipelines agent pool
+
+# Enable -Verbose option
+[CmdletBinding()]
+Param
+(
+	[Parameter(Mandatory=$true)]
+	[string]$Account,
+
+	[Parameter(Mandatory=$true)]
+	[String]$PersonalAccessToken,
+
+	[string]$AgentName = $env:COMPUTERNAME,
+
+	[string]$AgentNamePrefix= "dtl",
+
+	[Parameter(Mandatory=$true)]
+	[string]$PoolName,
+
+	[Parameter(Mandatory=$true)]
+	[int]$AgentCount
+)
+
+Write-Verbose "Entering powershell task" -verbose
+
+$currentLocation = Split-Path -parent $MyInvocation.MyCommand.Definition
+Write-Verbose "Current folder: $currentLocation" -verbose
+
+#Create a temporary directory where to download from VSTS the agent package (vsts-agent.zip) and then launch the configuration.
+$agentTempFolderName = Join-Path $env:temp ([System.IO.Path]::GetRandomFileName())
+New-Item -ItemType Directory -Force -Path $agentTempFolderName
+Write-Verbose "Temporary Agent download folder: $agentTempFolderName" -verbose
+
+$serverUrl = "https://dev.azure.com/$Account"
+Write-Verbose "Server URL: $serverUrl" -verbose
+
+$retryCount = 3
+$retries = 1
+Write-Verbose "Downloading Agent install files" -verbose
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+do
+{
+  try
+  {
+    Write-Verbose "Trying to get download URL for latest VSTS agent release..."
+    $vstsAgentUrl = "$serverUrl/_apis/distributedtask/packages/agent/win7-x64?`$top=1&api-version=3.0"
+    $basicAuth = (":{0}" -f  $PersonalAccessToken)
+    $basicAuth = [System.Text.Encoding]::UTF8.GetBytes($basicAuth)
+    $basicAuth = [System.Convert]::ToBase64String($basicAuth)
+    $headers = @{ Authorization = ("Basic {0}" -f $basicAuth) }
+    $agentList = Invoke-WebRequest -Uri $vstsAgentUrl -Headers $headers -Method Get -ContentType application/json -UseBasicParsing | ConvertFrom-Json
+    $agent = $agentList.value
+    if ($agent -is [Array])
+    {
+        $agent = $agentList.value[0]
+    }
+    $agentPackagePath = "$agentTempFolderName\agent.zip"
```

Comment:
If the file is partially created because the download fails, and we retry, will the existence of the file cause the subsequent call to download to fail or will the file be overwritten? From the logic, the temp folder would be the same when retrying. So, you may want to confirm. Just add a dummy `agent.zip` file in the folder and see if the call to download still works and overwrites as expected. Otherwise, in your `catch`, you may need to delete the `agent.zip` file before retrying.